Build error with Maven 3.5.2

#### Building with the latest maven (3.5.2) leads to a error while doing dependency resolution. The same source builds fine with maven 3.3.9.

Specifically,
```
[ERROR] The build could not read 2 projects -> [Help 1]
org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
[FATAL] 'dependencies.dependency. org.nd4j:nd4j-native:0.9.2-SNAPSHOT' for org.nd4j:nd4j-native:0.9.2-SNAPSHOT is referencing itself. @ org.nd4j:nd4j-native:[unknown-version], /Users/abhishek/code/skymind/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-native/pom.xml, line 18, column 21
[FATAL] 'dependencies.dependency. org.nd4j:nd4j-cuda-9.0:0.9.2-SNAPSHOT' for org.nd4j:nd4j-cuda-9.0:0.9.2-SNAPSHOT is referencing itself. @ org.nd4j:nd4j-cuda-9.0:[unknown-version], /Users/abhishek/code/skymind/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/pom.xml, line 173, column 21
```

Detailed error log [here](https://gist.github.com/abhishekpathak/440f630e27ce82e125137c1174fa7570).

--- a/nd4j-backends/nd4j-backend-impls/pom.xml
+++ b/nd4j-backends/nd4j-backend-impls/pom.xml
@@ -123,7 +123,7 @@
             </activation>
             <properties>
                 <dependency.groupId>${project.parent.groupId}</dependency.groupId>
-                <dependency.artifactId>${project.artifactId}</dependency.artifactId>
+                <dependency.artifactId>${project.parent.artifactId}</dependency.artifactId>
                 <dependency.version>${project.parent.version}</dependency.version>
                 <dependency.packaging>pom</dependency.packaging>
                 <dependency.classifier></dependency.classifier>

